### ISSUE TITLE
`no-unused-modules` should not halt execution if ignoreExports value is not found

### ISSUE THREAD

**Issue by Zamiell** [benmosher/eslint-plugin-import#2128]
I have an ESLint template config that contains the following:

```js
    "import/no-unused-modules": [
      "error",
      {
        missingExports: true,
        unusedExports: true,
        ignoreExports: ["src/**/*.d.ts"],
      },
    ],
 ```

This is because I want to use the `no-unused-modules` rule by default in my TypeScript projects, but I also want exclude any TypeScript definitions by default. (They will result in false positives.)

This mostly works great, but a big problem with this is that when I import the template config into a brand new TypeScript project, ESLint will refuse to execute, because no ".d.ts" files exist yet in the repository.

I think a safer default behavior here would be for ESLint to throw a warning and continue executing. Honestly, even showing a warning at all is pretty spurious in the case of a wildcard input, but I guess that could be argued either way.

**Comment by ljharb** [comment]
Unfortunately there's no consistent convention or expectation with star-glob paths that match nothing. Sometimes, what you *want* is what you're asking for - "0 or more" - but sometimes what you want is "1 or more", and you do indeed want an error when nothing matches.

In this case, it's an "ignore" setting, so I do think that it makes sense to be silent when nothing is found.
